src/testing/assemble_residual_on_subset_temporal_OOA.cpp

```diff
@@ -0,0 +1,298 @@
+            && this->all_parameters->ode_solver_param.atol == 1e-4 && this->all_parameters->ode_solver_param.rtol == 1e-4 
+            && this->all_parameters->time_refinement_study_param.number_of_times_to_solve == 1){
+            double L2_error_expected = 2.14808703658e-5; 
+            pcout << " Expected L2 error is: " << L2_error_expected << std::endl;
+            if (L2_error > L2_error_expected + 1e-14 || L2_error < L2_error_expected - 1e-14){
+                testfail = 1;
+                pcout << "Expected L2 error for RK3(2)5F[3S*+] using an atol = rtol = 1e-4 was not reached " << refinement <<std::endl;
+            }
+        }
+        
+        convergence_table.add_value("refinement", refinement);
+        convergence_table.add_value("dt", dt );
+        convergence_table.set_precision("dt", 16);
+        convergence_table.set_scientific("dt", true);
+        convergence_table.add_value("L1_error",L1_error);
+        convergence_table.set_precision("L1_error", 16);
+        convergence_table.evaluate_convergence_rates("L1_error", "dt", dealii::ConvergenceTable::reduction_rate_log2, 1);
+        convergence_table.add_value("L2_error",L2_error);
+        convergence_table.set_precision("L2_error", 16);
+        convergence_table.evaluate_convergence_rates("L2_error", "dt", dealii::ConvergenceTable::reduction_rate_log2, 1);
+        convergence_table.add_value("Linfty_error",Linfty_error);
+        convergence_table.set_precision("Linfty_error", 16);
+        convergence_table.evaluate_convergence_rates("Linfty_error", "dt", dealii::ConvergenceTable::reduction_rate_log2, 1);
+        
+        using ODESolverEnum = Parameters::ODESolverParam::ODESolverEnum;
+        if(params.ode_solver_param.ode_solver_type == ODESolverEnum::rrk_explicit_solver){
+            //for burgers, this is the average gamma over the runtime
+            const double gamma_aggregate_m1 = final_time_actual / (n_timesteps * dt)-1;
+            convergence_table.add_value("gamma_aggregate_m1", gamma_aggregate_m1);
+            convergence_table.set_precision("gamma_aggregate_m1", 16);
+            convergence_table.set_scientific("gamma_aggregate_m1", true);
+            convergence_table.evaluate_convergence_rates("gamma_aggregate_m1", "dt", dealii::ConvergenceTable::reduction_rate_log2, 1);
+        }
+ 
+        //Checking convergence order
+        const double expected_order = 2; // Hard-coded to order = 2
+        const double order_tolerance = 0.1;
+        if (refinement > 0) {
+            L2_error_conv_rate = -log(L2_error_old/L2_error)/log(refine_ratio);
+            pcout << "L2 order at " << refinement << " is " << L2_error_conv_rate << std::endl;
+            if (abs(L2_error_conv_rate - expected_order) > order_tolerance){
+                testfail = 1;
+                pcout << "Expected convergence order was not reached at refinement " << refinement <<std::endl;
+            }
+           
```

Comment:
Not sure how robust an OOA test is to small changes in the residual, such as skipping integration over a face. I think an OOA test should be good enough, but if time permits, one can also consider assembling the entire DG residual, then assembling DG residual for a group ID and then comparing the two vectors to see if the error in residual is close to machine precision.

Reply:
Thanks for the suggestion. I updated the unit tests to also assemble a residual without partitioning and compare the values. After some changes to account for ghost cells better, the updated test now passes on serial or MPI.